docker-compose.yaml

```diff
@@ -0,0 +1,130 @@
+version: "3.8"
+services:
+    ollama-server:
+        image: ollama/ollama:0.1.39
+        pull_policy: always
+        restart: always
+        container_name: ollama
+        ports:
+            - "${OLLAMA_PORT}:11434"
+        volumes:
+            - ./volumes/ollama:/root/.ollama
+        healthcheck:
+            test: ["CMD", "curl", "-f", "http://localhost:${OLLAMA_PORT}"]
+            interval: 1m
+            timeout: 10s
+            retries: 3
+        environment:
+            - OLLAMA_KEEP_ALIVE=${OLLAMA_KEEP_ALIVE}
+            - OLLAMA_HOST=${OLLAMA_HOST}
+        entrypoint: /bin/bash
+        command: -c "set -e; ollama serve & sleep 10 && ollama pull ${OLLAMA_MODEL} && sleep infinity"
+        networks:
+            - cognita-docker
+
+    infinity-server:
+        image: michaelf34/infinity:0.0.36
+        pull_policy: always
+        restart: always
+        container_name: infinity
+        ports:
+            - "${INFINITY_PORT}:7997"
+        volumes:
+            - ./volumes/infinity:/app/.cache
+        healthcheck:
+            test:
+                [
+                    "CMD",
+                    "curl",
+                    "-f",
+                    "http://localhost:${INFINITY_PORT}/health",
+                ]
+            interval: 1m
+            timeout: 10s
+            retries: 3
+        command: v2 --model-id ${INFINITY_EMBEDDING_MODEL} --model-id ${INFINITY_RERANKING_MODEL} --batch-size ${INFINITY_BATCH_SIZE}
+        networks:
+            - cognita-docker
+
+    qdrant-server:
+        image: qdrant/qdrant:v1.8.4
+        pull_policy: always
+        restart: always
+        container_name: qdrant
+        ports:
+            - "${QDRANT_REST}:6333"
+            - "${QDRANT_GRPC}:6334"
+        volumes:
+            - ./volumes/qdrant_storage:/qdrant/storage:z
```

Comment:
why does this need `z` mode?

Reply:
This was in qdrant docs, this is what I found:

:z: The SELinux label is set so that all containers can read/write the same volumes. This is typically used when multiple containers need to share a volume.